### PR TITLE
0.13: various integ test and logic fixes

### PR DESCRIPTION
**Pull request by Orzelius**
More info in commits

### CODE REVIEW

File: core/test/helpers.ts [hunk 1]
```diff
@@ -662,6 +663,24 @@ export function trimLineEnds(str: string) {
 
 const skipGroups = gardenEnv.GARDEN_SKIP_TESTS.split(" ")
 
+// Modified version of https://stackoverflow.com/a/26202058
+/**
+ * Recursively remove null on undefined values from an object (inluding arrays).
```
Comment (vvagaytsev) [inline core/test/helpers.ts:668]: Typo? on -> or

**Review by vvagaytsev — APPROVED** [review]
Great job! 🚀 Thank you! I've left one minor non-blocking comment :)

File: core/test/helpers.ts (hunk 1 above)
Comment (Orzelius) [reply core/test/helpers.ts]: good catch [578faf9](https://github.com/garden-io/garden/pull/3798/commits/578faf93acd77df7ade0964fd481e6cb8c4b2723)

**Comment by Orzelius** [comment]
Also took the chance to push [78efe83](https://github.com/garden-io/garden/pull/3798/commits/78efe8304edc44aad5ea0b25a5e1653516a22d69) which fixes all the tests other than the two volume one's failing for what seems internal solver issues

File: core/src/plugins/kubernetes/volumes/configmap.ts [hunk 2]
```diff
@@ -136,7 +133,6 @@ export const configMapModuleDefinition = (): ModuleTypeDefinition => ({
               dependencies: prepareRuntimeDependencies(module.spec.dependencies, dummyBuild),
 
               spec: {
-                accessModes: module.spec.accessModes,
```
Comment (edvald) [inline core/src/plugins/kubernetes/volumes/configmap.ts:139]: This seems incorrect. The issue is rather that [this](https://github.com/garden-io/garden/blob/33bef2a49828022ed8fed84a51823b3fd4d2c360/core/src/plugins/base-volume.ts#L20) is missing from [this](https://github.com/garden-io/garden/blob/212bbd3434d02b433cd078e32fb5ae668fbaff8a/core/src/plugins/kubernetes/volumes/configmap.ts#L33).

File: core/test/integ/src/plugins/kubernetes/container/deployment.ts [hunk 3]
```diff
@@ -659,7 +659,10 @@ describe("kubernetes container deployment handlers", () => {
         {
           name: "test",
           configMap: {
-            name: "configmap-module",
+            name: {
+              kind: "Deploy",
+              name: "configmap-module",
+            },
```
Comment (edvald) [inline core/test/integ/src/plugins/kubernetes/container/deployment.ts:668]: This doesn't seem right... I think the code being tested is wrong. Notice that we're validating a k8s Deployment resource, and there's definitely no place for `kind: "Deploy"` in there.

File: core/test/integ/src/plugins/kubernetes/container/deployment.ts [hunk 4]
```diff
@@ -427,6 +428,7 @@ describe("kubernetes container deployment handlers", () => {
 
     it("should configure the service for sync with sync mode enabled", async () => {
       const action = await resolveDeployAction("sync-mode")
+      await executeAction({ garden, graph, log: garden.log, action })
```
Comment (edvald) [inline core/test/integ/src/plugins/kubernetes/container/deployment.ts:431]: This shouldn't be necessary... What error came up without this?

**Review by edvald — CHANGES_REQUESTED** [review]

File: core/src/plugins/kubernetes/volumes/configmap.ts (hunk 2 above)
Comment (Orzelius) [reply core/src/plugins/kubernetes/volumes/configmap.ts:139]: I don't get it. There seems to be no such field on 0.12 and no logic to do anything if it's set, it also doesn't exist on the spec ts type. Am I missing something obvious?

File: core/test/integ/src/plugins/kubernetes/container/deployment.ts (hunk 3 above)
Comment (Orzelius) [reply core/test/integ/src/plugins/kubernetes/container/deployment.ts]: Good catch, fixed in [161d97f](https://github.com/garden-io/garden/pull/3798/commits/161d97f6706772aa76e15482e1a28b9f43586bb9)

File: core/test/integ/src/plugins/kubernetes/container/deployment.ts (hunk 4 above)
Comment (Orzelius) [reply core/test/integ/src/plugins/kubernetes/container/deployment.ts]: You're right. Removed in [ddf2bc4](https://github.com/garden-io/garden/pull/3798/commits/ddf2bc4e54678da25a6e52f026c4dc36e1091227).

File: core/src/plugins/kubernetes/volumes/configmap.ts (hunk 2 above)
Comment (edvald) [reply core/src/plugins/kubernetes/volumes/configmap.ts:139]: I'm not sure what to tell you... I linked to the field in my comment (also see BaseVolumeSpec), and I'm fairly sure I did not add this post 0.12 🤷‍♂️

Comment (Orzelius) [reply core/src/plugins/kubernetes/volumes/configmap.ts]: Restored the field and added schema extension in [4ffd0e2](https://github.com/garden-io/garden/pull/3798/commits/4ffd0e232e8faed9869e5ee72383aa6915a43248)

**Review by vvagaytsev — APPROVED** [review]

**Comment by Orzelius** [comment]
merging as all comments were addressed

**Comment by Orzelius** [comment]
dead link fix: https://github.com/garden-io/garden/pull/3851

**Pull request merged**